### PR TITLE
Update logic in the cache_refresh binary

### PR DESCRIPTION
**Pull request by dexter-1**
This commit updates the logic in the cache_refresh binary with the
following behaviors, for both users and groups:
  - If the metadata server returns successfully, but returns no users,
    create an empty cache file.
  - If the metadata server returns a 404 error, this means that OS Login
    is not enabled. Create a special error code for this and do not log an
    error if we see this code.
  - If the metadata server returns an error, don't update the cache file
    and log and error. If no cache file exists, add an empty one.

### CODE REVIEW

**Comment by illfelder** [comment]
This looks good to me. I'll wait for @hopkiw to review before merging.

File: src/cache_refresh/cache_refresh.cc
Comment (hopkiw) [inline src/cache_refresh/cache_refresh.cc:52]: no vanity functions, please. you can call stat in code where this was called.

File: src/cache_refresh/cache_refresh.cc
Comment (hopkiw) [inline src/cache_refresh/cache_refresh.cc:15]: why was this needed?

File: src/cache_refresh/cache_refresh.cc
Comment (hopkiw) [inline src/cache_refresh/cache_refresh.cc:85]: is the count variable now unused due to changes below?

File: src/cache_refresh/cache_refresh.cc
Comment (hopkiw) [inline src/cache_refresh/cache_refresh.cc:110]: i dont understand - if the real file exists, then remove the temporary file? how does it match the intention in the comment?

File: src/oslogin_utils.cc
Comment (hopkiw) [inline src/oslogin_utils.cc:208]: a 404 error was received

**Review by hopkiw — CHANGES_REQUESTED** [review]

File: src/oslogin_utils.cc
Comment (dexter-1) [reply src/oslogin_utils.cc]: Done.

File: src/cache_refresh/cache_refresh.cc
Comment (dexter-1) [reply src/cache_refresh/cache_refresh.cc]: Yes it is now unused. I removed it.

File: src/cache_refresh/cache_refresh.cc
Comment (dexter-1) [reply src/cache_refresh/cache_refresh.cc]: I'm not sure how that include got in there, removed it.

File: src/cache_refresh/cache_refresh.cc
Comment (dexter-1) [reply src/cache_refresh/cache_refresh.cc]: Done.

File: src/cache_refresh/cache_refresh.cc
Comment (dexter-1) [reply src/cache_refresh/cache_refresh.cc]: If the real file exists, we don't want to over-write it with the current temporary file on a server error. So we should remove the temporary file and return here, so we don't execute the code after which tries to move the temporary file to the real file. I've updated the comment, I hope that makes it less confusing.

Comment (hopkiw) [reply src/cache_refresh/cache_refresh.cc]: i missed that this block ends in 'return', it makes sense now

**Pull request merged**
